nodejs/init.sls

```diff
@@ -1,10 +1,20 @@
-node-pkgs:
-  pkg:
-    - installed
-    - names:
-      - npm
-      - nodejs-legacy
+{% import 'project/_vars.sls' as vars with context %}
+include:
+  - project.web.app
 
-# note: 'nodejs-legacy' is an ubuntu package that installs node then adds
-# a symlink to /usr/bin/node, which is where every other Linux distribution
-# puts the node executable.
+# Install NodeJS v4.x, to match local frontend dev setup
+nodejs_repo:
+  pkgrepo.managed:
+    - humanname: NodeSource Node.js PPA
+    - name: deb https://deb.nodesource.com/node_4.x {{ grains['lsb_distrib_codename'] }} main
+    - file: /etc/apt/sources.list.d/nodesource.list
+    - key_url: salt://project/web/nodesource.pub
+    - require_in:
+      - pkg: nodejs
+
+nodejs:
+  pkg.removed:
+    - name: npm
+    - name: nodejs-legacy
+  pkg.latest:
+    - name: nodejs
```

Comment:
Apparently this isn't quite valid... I would have thought it'd work like this.  Maybe try separating these two states?

[127.0.0.1:2222] sudo: salt -G 'environment:local' -linfo state.highstate 
[127.0.0.1:2222] out: vagrant-ubuntu-trusty-64:
[127.0.0.1:2222] out:     Data failed to compile:
[127.0.0.1:2222] out: ----------
[127.0.0.1:2222] out:     ID 'nodejs' in SLS 'nodejs' contains multiple state declarations of the same type